src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheCrusader/boss_lord_jaraxxus.cpp

```diff
@@ -106,23 +122,53 @@ class boss_jaraxxus : public CreatureScript
 
             void Reset() override
             {
-                _Reset();
-                events.ScheduleEvent(EVENT_FEL_FIREBALL, 5*IN_MILLISECONDS);
-                events.ScheduleEvent(EVENT_FEL_LIGHTNING, urand(10*IN_MILLISECONDS, 15*IN_MILLISECONDS));
-                events.ScheduleEvent(EVENT_INCINERATE_FLESH, urand(20*IN_MILLISECONDS, 25*IN_MILLISECONDS));
-                events.ScheduleEvent(EVENT_NETHER_POWER, 40*IN_MILLISECONDS);
-                events.ScheduleEvent(EVENT_LEGION_FLAME, 30*IN_MILLISECONDS);
-                events.ScheduleEvent(EVENT_SUMMONO_NETHER_PORTAL, 20*IN_MILLISECONDS);
-                events.ScheduleEvent(EVENT_SUMMON_INFERNAL_ERUPTION, 80*IN_MILLISECONDS);
+                events.Reset();
+                summons.DespawnAll();
+                me->SetCombatPulseDelay(0);
+                me->ResetLootMode();
+                if (instance->GetBossState(DATA_JARAXXUS) == NOT_STARTED)
+                    DoAction(ACTION_JARAXXUS_INTRO);
             }
 
             void JustReachedHome() override
             {
                 _JustReachedHome();
                 instance->SetBossState(DATA_JARAXXUS, FAIL);
-                DoCast(me, SPELL_JARAXXUS_CHAINS);
-                me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE);
+                DoCastSelf(SPELL_JARAXXUS_CHAINS);
                 me->SetImmuneToPC(true);
+                me->SetReactState(REACT_PASSIVE);
+            }
+
+            void DoAction(int32 action) override
+            {
+                if (action == ACTION_JARAXXUS_INTRO)
+                {
+                    me->SetReactState(REACT_PASSIVE);
+                    events.SetPhase(PHASE_INTRO);
+                    events.ScheduleEvent(EVENT_INTRO, Seconds(1));
+                }
+                if (action == ACTION_JARAXXUS_ENGAGE)
```

Comment:
please add a "else" here or use a switch, up to you

Reply:
yeah, i forgot this one :)
Fixed.